lib/dashboard/widget/StrictGraphWidget.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  GraphWidget,
+  HorizontalAnnotation,
+  IMetric,
+  LegendPosition,
+  YAxisProps,
+} from "aws-cdk-lib/aws-cloudwatch";
+
+export interface SingleAxisGraphWidgetProps {
+  readonly title?: string;
+  readonly width: number;
+  readonly height: number;
+  readonly metrics: IMetric[];
+  readonly axis: YAxisProps;
+  readonly annotations?: HorizontalAnnotation[];
+}
+
+/**
+ * Line graph widget with one axis only (left).
+ * If there is just one metric, it will hide the legend to save space.
+ * The purpose of this custom class is to make the properties more strict.
+ * It will avoid graphs with no axis, and so on.
+ */
+export class SingleAxisGraphWidget extends GraphWidget {
+  constructor(props: SingleAxisGraphWidgetProps) {
+    super(SingleAxisGraphWidget.transformProps(props));
+  }
+
+  private static transformProps(props: SingleAxisGraphWidgetProps) {
```

Comment:
Nitpick: add a return type. I should probably get around to revisiting our linting setup.

Reply:
Will do. As for the linting, there is even a ticket for it, since I was not sure what linting config is actually the one used - might be a projen configuration issue.